pkg/splunk/enterprise/util.go

```diff
@@ -964,3 +1127,170 @@ func initAndCheckAppInfoStatus(client splcommon.ControllerClient, cr splcommon.M
 
 	return nil
 }
+
+//////////////////////////
+//
+// App Framework Action Functions
```

Comment:
This need a new home too to isolate these actions.  With this method too its fairly easy to see how we could implement a disable/delete as well.